Week3/api.js

```diff
@@ -0,0 +1,218 @@
+const mysql = require('mysql');
+const util = require('util');
+const { handelError } = require('./dbUtils');
+
+const connection = mysql.createConnection({
+  host: 'localhost',
+  user: 'hyfuser',
+  password: 'hyfpassword',
+  database: 'todos_app',
+});
+
+const execQuery = util.promisify(connection.query.bind(connection));
+connection.connect(handelError('connecting...'));
+
+class Api {
+  async getUsersHandler(req, res) {
+    try {
+      const users = await execQuery(`SELECT * FROM todos_app.users`);
+      if (!users.length) return res.status(404).send(`You need to insert some users!`);
+      res.send(users);
+    } catch (err) {
+      console.log(err.message);
```

Comment:
If you get an error you have to return an error code (500) so the client knows something wrong happened.